### DISCUSSION TITLE
Following instructions for quarto publish gh-pages - "fatal: a branch named 'gh-pages' already exists"

### DISCUSSION THREAD

**Discussion by elsherbini** [quarto-dev/quarto-cli#6972] [Q&A]
### Description

I'm trying to publish a website on gh-pages. I'm using windows and the repo is on a OneDrive directory. I usually use ubuntu, so I'm not sure all the implications of working on a OneDrive directory. *Edit: I just tried doing it from scratch in a non-OneDrive directory and got the same error, so I don't think that's the problem*

here is the repo, just taking from andrewpbrays site for posit::conf 2023 to test deploying:
https://github.com/elsherbini/workshop_website_test

I followed the instructions on https://quarto.org/docs/publishing/github-pages.html#source-branch to set up gh-pages to make the orphan gh-pages branch, and then checkout the main branch again and tried "quarto publish gh-pages"

```
C:\Users\kwonl\OneDrive\Desktop\projects\positconf_test>quarto publish gh-pages
? Publish site to https://elsherbini.github.io/workshop_website_test/ using gh-pages? (Y/n) » Yes
fatal: a branch named 'gh-pages' already exists
Your branch is up to date with 'origin/main'.
Already on 'main'
On branch main
Your branch is up to date with 'origin/main'.
```

Googling around, some people not using quarto suggested manually deleting `node_modules/gh-pages/.cache`, but I've searched everywhere for that and can't find it.

**Comment by elsherbini** [comment]
I had success finally. I had to remove the gh-pages branch from both local and remote (in spite of the instructions saying to make the orphan branch manually) using `git branch -D gh-pages` and `git push -d origin gh-pages` . I then had to start the ssh-agent - git push worked but when quarto tried to do it I got the following:

```
C:\Users\kwonl\Documents\workshop_website>quarto publish gh-pages
? Publish site to https://elsherbini.github.io/workshop2023/ using gh-pages? (Y/n) » Yes
Switched to a new branch 'gh-pages'
[gh-pages (root-commit) 9709d82] Initializing gh-pages branch
ssh_askpass: exec(/usr/lib/ssh/ssh-askpass): No such file or directory
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.
```

So I ran `start-ssh-agent` because I'm on windows, and `ssh-add path\to\id_rsa`, then deleted the failed branch again, and ran quarto publish gh-pages:

```
C:\Users\kwonl\Documents\workshop_website>quarto publish gh-pages
? Publish site to https://elsherbini.github.io/workshop2023/ using gh-pages? (Y/n) » Yes
Switched to a new branch 'gh-pages'
[gh-pages (root-commit) 25bb6d6] Initializing gh-pages branch
remote:
remote: Create a pull request for 'gh-pages' on GitHub by visiting:
remote:      https://github.com/elsherbini/workshop2023/pull/new/gh-pages
remote:
To github.com:elsherbini/workshop2023.git
 * [new branch]      HEAD -> gh-pages
Your branch is up to date with 'origin/main'.
Switched to branch 'main'
From github.com:elsherbini/workshop2023
 * branch            gh-pages   -> FETCH_HEAD
Rendering for publish:
...
origin  git@github.com:elsherbini/workshop2023.git (push)
To github.com:elsherbini/workshop2023.git
   25bb6d6..15dd04a  HEAD -> gh-pages

[>] Deploying gh-pages branch to website (this may take a few minutes)
[>] Published to https://elsherbini.github.io/workshop2023/
```

**Comment by mcanouil** [comment]
These are known current documentation issues tracked:
- https://github.com/quarto-dev/quarto-cli/issues/6204
